app/helpers/manipulation_helper.rb

```diff
@@ -0,0 +1,14 @@
+module ManipulationHelper
+  def manipulate_storage_systems(assoc, storage_systems = nil)
```

Comment:
Naming wise, I'm not sure the term manipulate is clear here.  This seems to just generate a where clause, so perhaps

```suggestion
  def storage_systems_where_clause(assoc, storage_systems = nil)
```